<a href="https://colab.research.google.com/github/urviparikh/DATA-690-WANG/blob/main/Assignment_09/Assignment_09.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

 **Assignment 9**
 
**Author: Urvi Parikh**

In [41]:
import pandas as pd
import numpy as np
import scipy.stats as st
from scipy.stats import t
import math

A random survey of enrollment at 35 community colleges across the United States yielded the following figures:

6,414; 1,550; 2,109; 9,350; 21,828; 4,300; 5,944; 5,722; 2,825; 2,044;

5,481; 5,200; 5,853; 2,750; 10,012; 6,357; 27,000; 9,414; 7,681; 3,200;

17,500; 9,200; 7,380; 18,314; 6,557; 13,713; 17,768; 7,493; 2,771; 2,861;

1,263; 7,285; 28,165; 5,080; 11,622

**Step 0 - Data Preprocessing**

In [42]:
# make each line of numbers a string object and then concatenate them together 
# The end result is one single string containing 35 numbers separated by ";"

data_1 = "6,414; 1,550; 2,109; 9,350; 21,828; 4,300; 5,944; 5,722; 2,825; 2,044;" 
data_2 = "5,481; 5,200; 5,853; 2,750; 10,012; 6,357; 27,000; 9,414; 7,681; 3,200; "
data_3 = "17,500; 9,200; 7,380; 18,314; 6,557; 13,713; 17,768; 7,493; 2,771; 2,861; "
data_4 = "1,263; 7,285; 28,165; 5,080; 11,622"
data = data_1 + data_2 + data_3 + data_4
print(data)

6,414; 1,550; 2,109; 9,350; 21,828; 4,300; 5,944; 5,722; 2,825; 2,044;5,481; 5,200; 5,853; 2,750; 10,012; 6,357; 27,000; 9,414; 7,681; 3,200; 17,500; 9,200; 7,380; 18,314; 6,557; 13,713; 17,768; 7,493; 2,771; 2,861; 1,263; 7,285; 28,165; 5,080; 11,622


In [43]:
# Convert the single string to a list of strings using split() function
# Make sure to specify a delimter or separator

my_string = data.split(';')
print(my_string)

['6,414', ' 1,550', ' 2,109', ' 9,350', ' 21,828', ' 4,300', ' 5,944', ' 5,722', ' 2,825', ' 2,044', '5,481', ' 5,200', ' 5,853', ' 2,750', ' 10,012', ' 6,357', ' 27,000', ' 9,414', ' 7,681', ' 3,200', ' 17,500', ' 9,200', ' 7,380', ' 18,314', ' 6,557', ' 13,713', ' 17,768', ' 7,493', ' 2,771', ' 2,861', ' 1,263', ' 7,285', ' 28,165', ' 5,080', ' 11,622']


Create a list of integers from the list of strings using List Comprehension or for loop. Make sure to remove the "," first and then convert the strings to integers.

In [44]:
my_list = []

for i in my_string:
    string1 = i.replace(',', '')
    string2 = int(string1)

    my_list.append(string2)

print(my_list)


[6414, 1550, 2109, 9350, 21828, 4300, 5944, 5722, 2825, 2044, 5481, 5200, 5853, 2750, 10012, 6357, 27000, 9414, 7681, 3200, 17500, 9200, 7380, 18314, 6557, 13713, 17768, 7493, 2771, 2861, 1263, 7285, 28165, 5080, 11622]


**Step 1 - Calculate and Display the Sample Size and Sample Mean**

In [45]:
# Calculate and display the sample size

sample_size = len(my_list)

print("Sample size =", sample_size)

Sample size = 35


In [46]:
# Calculate and display the sample mean

sample_mean = sum(my_list)/len(my_list)
sample_mean = float(math.ceil(sample_mean))

print("Sample mean = ", sample_mean)

Sample mean =  8629.0


The point estimate of the mean enrollment of US community colleges is **8629**.

**Step 2 - Calculate and Display the Sample Standard Deviation & Sample Standard Error**


Sample Standard Deviation $S=\sqrt{\dfrac{1}{n-1}\sum\limits_{i=1}^n (X_i-\bar{X})^2}$

Sample Standard Error = $\dfrac{S}{\sqrt{n}}$

Note: The default Delta Degree of Freedom (DDOF) for Numpy's std function is 0 which is applicable to populate data. For sample data, we need to specify ddof=1.

For the enrollment data, we round up the statistics to be the full integers (no decimal points).

In [49]:
# Calculate and display the sample standard deviation using Numpy's std function

std_dev = round(np.std(my_list, ddof=1))

print("Sample Standard Deviation = ", std_dev)

Sample Standard Deviation =  6944


In [50]:
# Calculate and display the sample standard error

std_error = round(std_dev / np.sqrt(sample_size))

print("Sample standard error is ", std_error)


Sample standard error is  1174


**Step 3 - Calculate t Critical Value using t-Distribution**


$\alpha$ = 1 - Confidence Level = 1 - 95% = 0.05

$\dfrac{\alpha}{2}$ = 0.025

n (sample size) = 35

df (degree of freedom) = n - 1 = 35 - 1 = 34

We will use Python scipy.stats t-distribution's PPF (Percentage Point Function) to calculate t critical value $t_{0.025,34}$.

In [52]:
# Calculate and display the t critical value using scipy.stats.t package ppf function

perc_conf_req = .95
q = ((1-perc_conf_req)/2)
n = sample_size
df = n - 1


t = st.t.ppf(q, df)
t = round(abs(t),2)

print("t critical value is ", t)


t critical value is  2.03


**Step 4 - Calculate the Margin of Error**


Margin of Error = t-Statistics * Sample Standard Error = $t_{\alpha/2,n-1}\left(\dfrac{s}{\sqrt{n}}\right)$

In [53]:
# Calculate and display the margin of error

Margin_error = std_error * t
Margin_error = float(math.floor(Margin_error))

print("Margin of error = ", Margin_error)

Margin of error =  2383.0


**Step 5 - Calculate Lower and Upper Limit of the Confidence Interval**


Lower Limit = Sample Mean - Margin of Error

Upper Limit = Sample Mean + Margin of Error

In [54]:
# Calculate and display the lower limit

lower_limit = sample_mean - Margin_error

print("Lower limit is ",lower_limit)

Lower limit is  6246.0


In [55]:
# Calculate and display the upper limit

upper_limit = sample_mean + Margin_error

print("Upper limit is ", upper_limit)

Upper limit is  11012.0


**Step 6 - Now We have the 95% Confidence Interval**


Confidence Interval ($\sigma$ unknown) = $\bar{x}\space\pm\space t_{\alpha/2}\left(\dfrac{S}{\sqrt{n}}\right)$ = Sample_Mean $\pm$ Margin of Error

In [56]:
# Print the 95% confidence interveral in the form (lower limit, upper limit)

print("The 95% confidence interval = ",(lower_limit, upper_limit))

The 95% confidence interval =  (6246.0, 11012.0)
